Error on update aws route53 a record to cname record

version: Terraform v0.7.7

I want to change A record to CNAME record with the same hostname, like

FROM:

```
resource "aws_route53_record" "www" {
   zone_id = "${aws_route53_zone.primary.zone_id}"
   name = "www.example.com"
   type = "A"
   ttl = "300"
   records = ["${aws_eip.lb.public_ip}"]
}
```

TO:

```
resource "aws_route53_record" "www" {
   zone_id = "${aws_route53_zone.primary.zone_id}"
   name = "www.example.com"
   type = "CNAME"
   ttl = "300"
   records = ["elb-address"]
}
```

Error happened:

```
1 error(s) occurred:

* aws_route53_record.cname_record: [ERR]: Error building changeset: InvalidChangeBatch: RRSet of type CNAME with DNS name www.example.com is not permitted as it conflicts with other records with the same DNS name in zone xxxx.
	status code: 400, request id: 012b1971-9f3a-11e6-88df-7b2406432013
```

Maybe the new resource is created before the old one is deleted.

@kwilczynski I also tried the reversed way, from cname to a record with the same dns name, the same error happend. and below is the debug log for the reversed way

```
2016/10/31 18:01:05 [INFO] Terraform version: 0.7.7  fa6a83ebdc323f2b415779786e102e69ddbf9a48
2016/10/31 18:01:05 [INFO] CLI args: []string{"/usr/local/bin/terraform", "apply", "tf_test_env.tfout"}
2016/10/31 18:01:05 [DEBUG] Detected home directory from env var: /Users/hanks
2016/10/31 18:01:05 [DEBUG] Detected home directory from env var: /Users/hanks
2016/10/31 18:01:05 [DEBUG] Attempting to open CLI config file: /Users/hanks/.terraformrc
2016/10/31 18:01:05 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2016/10/31 18:01:05 [DEBUG] Detected home directory from env var: /Users/hanks
2016/10/31 18:01:05 [INFO] Validating the context...
2016/10/31 18:01:05 [DEBUG] DestroyEdgeInclude: Checking: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] DestroyEdgeInclude: Checking: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] DestroyEdgeInclude: Checking: module.test_record.aws_route53_record.a_record
iable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
odeConfigVariable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.aws_region
2016/10/31 18:01:05 [DEBUG] Found up edge to provider.aws (disabled), var is not noop
2016/10/31 18:01:05 [DEBUG] Checking resource noop: aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] Diff has aws_route53_record.a_record, resource is not a noop
2016/10/31 18:01:05 [DEBUG] Final noop value: false
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.ip_list
2016/10/31 18:01:05 [DEBUG] Found up edge to module.test_record.aws_route53_record.a_record, var is not noop
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.name
2016/10/31 18:01:05 [DEBUG] Found up edge to module.test_record.aws_route53_record.a_record, var is not noop
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.aws_secret_key
2016/10/31 18:01:05 [DEBUG] Found up edge to provider.aws (disabled), var is not noop
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.zone_id
2016/10/31 18:01:05 [DEBUG] Found up edge to module.test_record.aws_route53_record.a_record, var is not noop
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.aws_access_key
2016/10/31 18:01:05 [DEBUG] Found up edge to provider.aws (disabled), var is not noop
raphNodeConfigOutputFlat
var.aws_access_key - *terraform.GraphNodeConfigVariable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
m.GraphNodeConfigOutputFlat
var.aws_access_key - *terraform.GraphNodeConfigVariable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
form.GraphNodeConfigOutputFlat
var.aws_access_key - *terraform.GraphNodeConfigVariable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
onfigOutputFlat
  module.test_record.plan-destroy - *terraform.graphNodeModuleDestroyFlat
  provider.module.test_record.aws (close) - *terraform.graphNodeCloseProvider
var.aws_access_key - *terraform.GraphNodeConfigVariable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
2016/10/31 18:01:05 [DEBUG] Starting graph walk: walkValidate
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.name: walking
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.name: evaluating
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_access_key: walking
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.ip_list: walking
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_region: walking
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_access_key: evaluating
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_region: evaluating
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.ip_list: evaluating
2016/10/31 18:01:05 [TRACE] [walkValidate] Entering eval tree: var.aws_region
2016/10/31 18:01:05 [TRACE] [walkValidate] Entering eval tree: module.test_record.var.ip_list
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.plan-destroy: walking
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.zone_id: walking
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_secret_key: walking
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_secret_key: evaluating
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.zone_id: evaluating
2016/10/31 18:01:05 [DEBUG] vertex provider.aws (disabled), got dep: var.aws_region
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalCoerceMapVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalTypeCheckVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSetVariables
2016/10/31 18:01:05 [TRACE] [walkValidate] Entering eval tree: var.aws_secret_key
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] vertex provider.aws (disabled), got dep: var.aws_access_key
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.plan-destroy: evaluating
2016/10/31 18:01:05 [DEBUG] vertex provider.aws (disabled), got dep: var.aws_secret_key
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] vertex root.provider.aws (disabled): walking
2016/10/31 18:01:05 [DEBUG] vertex root.provider.aws (disabled): evaluating
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalOpFilter
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalVariableBlock
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalCoerceMapVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalTypeCheckVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSetVariables
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalVariableBlock
2016/10/31 18:01:05 [TRACE] [walkValidate] Exiting eval tree: module.test_record.var.ip_list
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalCoerceMapVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalTypeCheckVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSetVariables
2016/10/31 18:01:05 [DEBUG] vertex module.test_record.aws_route53_record.a_record, got dep: module.test_record.var.name
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalBuildProviderConfig
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSetProviderConfig
2016/10/31 18:01:05 [DEBUG] vertex module.test_record.provider.aws, got dep: provider.aws (disabled)
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.provider.aws: walking
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.provider.aws: evaluating
2016/10/31 18:01:05 [DEBUG] plugin: starting plugin: /usr/local/bin/terraform []string{"/usr/local/bin/terraform", "internal-plugin", "provider", "aws"}
2016/10/31 18:01:05 [DEBUG] plugin: waiting for RPC address for: /usr/local/bin/terraform
2016/10/31 18:01:05 [DEBUG] plugin: terraform: 2016/10/31 18:01:05 [INFO] Terraform version: 0.7.7  fa6a83ebdc323f2b415779786e102e69ddbf9a48
2016/10/31 18:01:05 [DEBUG] plugin: terraform: 2016/10/31 18:01:05 [INFO] CLI args: []string{"/usr/local/bin/terraform", "internal-plugin", "provider", "aws"}
2016/10/31 18:01:05 [DEBUG] plugin: terraform: 2016/10/31 18:01:05 [DEBUG] Detected home directory from env var: /Users/hanks
2016/10/31 18:01:05 [DEBUG] plugin: terraform: 2016/10/31 18:01:05 [DEBUG] Detected home directory from env var: /Users/hanks
2016/10/31 18:01:05 [DEBUG] plugin: terraform: 2016/10/31 18:01:05 [DEBUG] Attempting to open CLI config file: /Users/hanks/.terraformrc
2016/10/31 18:01:05 [DEBUG] plugin: terraform: 2016/10/31 18:01:05 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2016/10/31 18:01:05 [DEBUG] plugin: terraform: 2016/10/31 18:01:05 [DEBUG] Detected home directory from env var: /Users/hanks
2016/10/31 18:01:05 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:05 [INFO] Starting provider plugin aws
2016/10/31 18:01:05 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:05 [DEBUG] plugin: plugin address: unix /var/folders/fs/zn9snpfj31l50sv981ngz_lr0000gn/T/plugin642154693
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalOpFilter
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalGetProvider
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalBuildProviderConfig
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalValidateProvider
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSetProviderConfig
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [TRACE] [walkValidate] Exiting eval tree: module.test_record.provider.aws
2016/10/31 18:01:05 [DEBUG] vertex module.test_record.aws_route53_record.cname_record (orphan), got dep: module.test_record.provider.aws
2016/10/31 18:01:05 [DEBUG] vertex module.test_record.aws_route53_record.a_record, got dep: module.test_record.provider.aws
2016/10/31 18:01:05 [DEBUG] vertex provider.module.test_record.aws (close), got dep: module.test_record.provider.aws
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.aws_route53_record.cname_record (orphan): walking
2016/10/31 18:01:05 [DEBUG] vertex module.test_record.aws_route53_record.a_record, got dep: module.test_record.var.ip_list
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.aws_route53_record.a_record: walking
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.aws_route53_record.cname_record (orphan): evaluating
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.aws_route53_record.a_record: evaluating
2016/10/31 18:01:05 [TRACE] [walkValidate] Entering eval tree: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:05 [TRACE] [walkValidate] Entering eval tree: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalInstanceInfo
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] vertex provider.module.test_record.aws (close), got dep: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalOpFilter
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalValidateCount
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalCountFixZeroOneBoundary
2016/10/31 18:01:05 [TRACE] [walkValidate] Exiting eval tree: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.aws_route53_record.a_record: expanding/walking dynamic subgraph
2016/10/31 18:01:05 [DEBUG] vertex root.test_record.aws_route53_record.a_record: walking
2016/10/31 18:01:05 [DEBUG] vertex root.test_record.aws_route53_record.a_record: evaluating
2016/10/31 18:01:05 [TRACE] [walkValidate] Entering eval tree: aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalOpFilter
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalGetProvider
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalValidateResource
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalInstanceInfo
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [TRACE] [walkValidate] Exiting eval tree: aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] vertex module.test_record.output.fqdn, got dep: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] vertex provider.module.test_record.aws (close), got dep: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.output.fqdn: walking
2016/10/31 18:01:05 [DEBUG] vertex root.provider.module.test_record.aws (close): walking
2016/10/31 18:01:05 [DEBUG] vertex root.provider.module.test_record.aws (close): evaluating
2016/10/31 18:01:05 [TRACE] [walkValidate] Entering eval tree: provider.module.test_record.aws (close)
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalCloseProvider
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.output.fqdn: evaluating
2016/10/31 18:01:05 [TRACE] [walkValidate] Exiting eval tree: provider.module.test_record.aws (close)
2016/10/31 18:01:05 [TRACE] [walkValidate] Entering eval tree: module.test_record.output.fqdn
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalOpFilter
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalWriteOutput
2016/10/31 18:01:05 [DEBUG] vertex root, got dep: module.test_record.output.fqdn
2016/10/31 18:01:05 [DEBUG] vertex root, got dep: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:05 [DEBUG] vertex root, got dep: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] vertex root, got dep: module.test_record.plan-destroy
2016/10/31 18:01:05 [DEBUG] vertex root, got dep: provider.module.test_record.aws (close)
2016/10/31 18:01:05 [DEBUG] vertex root.root: walking
2016/10/31 18:01:05 [INFO] Validation result: 0 warnings, 0 errors
2016/10/31 18:01:05 [DEBUG] vertex module.test_record: static expanding
2016/10/31 18:01:05 [DEBUG] DestroyEdgeInclude: Checking: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] DestroyEdgeInclude: Checking: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] DestroyEdgeInclude: Checking: module.test_record.aws_route53_record.a_record
iable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
odeConfigVariable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.name
2016/10/31 18:01:05 [DEBUG] Found up edge to module.test_record.aws_route53_record.a_record, var is not noop
2016/10/31 18:01:05 [DEBUG] Checking resource noop: aws_route53_record.a_record
2016/10/31 18:01:05 [DEBUG] Diff has aws_route53_record.a_record, resource is not a noop
2016/10/31 18:01:05 [DEBUG] Final noop value: false
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.ip_list
2016/10/31 18:01:05 [DEBUG] Found up edge to module.test_record.aws_route53_record.a_record, var is not noop
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.aws_secret_key
2016/10/31 18:01:05 [DEBUG] Found up edge to provider.aws (disabled), var is not noop
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.aws_access_key
2016/10/31 18:01:05 [DEBUG] Found up edge to provider.aws (disabled), var is not noop
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.zone_id
2016/10/31 18:01:05 [DEBUG] Found up edge to module.test_record.aws_route53_record.a_record, var is not noop
2016/10/31 18:01:05 [DEBUG] Checking variable noop: var.aws_region
2016/10/31 18:01:05 [DEBUG] Found up edge to provider.aws (disabled), var is not noop
raphNodeConfigOutputFlat
var.aws_access_key - *terraform.GraphNodeConfigVariable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
m.GraphNodeConfigOutputFlat
var.aws_access_key - *terraform.GraphNodeConfigVariable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
form.GraphNodeConfigOutputFlat
var.aws_access_key - *terraform.GraphNodeConfigVariable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
onfigOutputFlat
  module.test_record.plan-destroy - *terraform.graphNodeModuleDestroyFlat
  provider.module.test_record.aws (close) - *terraform.graphNodeCloseProvider
var.aws_access_key - *terraform.GraphNodeConfigVariable
var.aws_region - *terraform.GraphNodeConfigVariable
var.aws_secret_key - *terraform.GraphNodeConfigVariable
2016/10/31 18:01:05 [DEBUG] Starting graph walk: walkApply
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.name: walking
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_region: walking
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_region: evaluating
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.name: evaluating
2016/10/31 18:01:05 [TRACE] [walkApply] Entering eval tree: module.test_record.var.name
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.plan-destroy: walking
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_secret_key: walking
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_secret_key: evaluating
2016/10/31 18:01:05 [TRACE] [walkApply] Entering eval tree: var.aws_secret_key
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.zone_id: walking
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.zone_id: evaluating
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [TRACE] [walkApply] Entering eval tree: var.aws_region
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalNoop
2016/10/31 18:01:05 [TRACE] [walkApply] Exiting eval tree: var.aws_region
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.plan-destroy: evaluating
2016/10/31 18:01:05 [DEBUG] vertex root.var.aws_access_key: evaluating
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalCoerceMapVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalTypeCheckVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSetVariables
2016/10/31 18:01:05 [TRACE] [walkApply] Exiting eval tree: var.aws_secret_key
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.ip_list: walking
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.var.ip_list: evaluating
2016/10/31 18:01:05 [TRACE] [walkApply] Entering eval tree: module.test_record.var.ip_list
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] vertex provider.aws (disabled), got dep: var.aws_region
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] vertex provider.aws (disabled), got dep: var.aws_access_key
2016/10/31 18:01:05 [DEBUG] vertex root.provider.aws (disabled): walking
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalVariableBlock
2016/10/31 18:01:05 [DEBUG] vertex root.provider.aws (disabled): evaluating
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalCoerceMapVariable
2016/10/31 18:01:05 [TRACE] [walkApply] Entering eval tree: provider.aws (disabled)
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalTypeCheckVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalOpFilter
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSetVariables
2016/10/31 18:01:05 [TRACE] [walkApply] Exiting eval tree: module.test_record.var.zone_id
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalVariableBlock
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalCoerceMapVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalTypeCheckVariable
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSetVariables
2016/10/31 18:01:05 [DEBUG] vertex module.test_record.aws_route53_record.a_record, got dep: module.test_record.var.ip_list
2016/10/31 18:01:05 [DEBUG] vertex module.test_record.aws_route53_record.a_record, got dep: module.test_record.var.zone_id
2016/10/31 18:01:05 [DEBUG] vertex module.test_record.aws_route53_record.a_record, got dep: module.test_record.var.name
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalBuildProviderConfig
2016/10/31 18:01:05 [DEBUG] root: eval: *terraform.EvalSetProviderConfig
2016/10/31 18:01:05 [TRACE] [walkApply] Exiting eval tree: provider.aws (disabled)
2016/10/31 18:01:05 [DEBUG] vertex module.test_record.provider.aws, got dep: provider.aws (disabled)
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.provider.aws: walking
2016/10/31 18:01:05 [DEBUG] vertex root.module.test_record.provider.aws: evaluating
2016/10/31 18:01:05 [TRACE] [walkApply] Entering eval tree: module.test_record.provider.aws
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalInitProvider
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalOpFilter
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalGetProvider
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalInterpolate
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalBuildProviderConfig
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSetProviderConfig
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalOpFilter
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:05 [DEBUG] root.test_record: eval: *terraform.EvalConfigProvider
2016/10/31 18:01:05 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:05 [INFO] No assume_role block read from configuration
2016/10/31 18:01:05 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:05 [INFO] Building AWS region structure
2016/10/31 18:01:05 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:05 [INFO] Building AWS auth structure
2016/10/31 18:01:05 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:05 [WARN] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
2016/10/31 18:01:05 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:05 [INFO] AWS Auth provider used: "StaticProvider"
2016/10/31 18:01:05 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:05 [DEBUG] [aws-sdk-go] DEBUG: Request sts/GetCallerIdentity Details:
2016/10/31 18:01:05 [DEBUG] plugin: terraform: ---[ REQUEST POST-SIGN ]-----------------------------
2016/10/31 18:01:05 [DEBUG] plugin: terraform: POST / HTTP/1.1
2016/10/31 18:01:05 [DEBUG] plugin: terraform: Host: sts.amazonaws.com
2016/10/31 18:01:05 [DEBUG] plugin: terraform: User-Agent: terraform/0.7.7 aws-sdk-go/1.4.14 (go1.7.1; darwin; amd64)
2016/10/31 18:01:05 [DEBUG] plugin: terraform: Content-Length: 43
2016/10/31 18:01:05 [DEBUG] plugin: terraform: Authorization: AWS4-HMAC-SHA256 Credential=AKIAIUVMU4M7NKKFBJBQ/20161031/us-east-1/sts/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date, Signature=9398ca57ec43a32af8f76fd87d143d9464a90c2566080b54f30ef463ef0c4029
2016/10/31 18:01:05 [DEBUG] plugin: terraform: Content-Type: application/x-www-form-urlencoded; charset=utf-8
2016/10/31 18:01:05 [DEBUG] plugin: terraform: X-Amz-Date: 20161031T090105Z
2016/10/31 18:01:05 [DEBUG] plugin: terraform: Accept-Encoding: gzip
2016/10/31 18:01:05 [DEBUG] plugin: terraform: 
2016/10/31 18:01:05 [DEBUG] plugin: terraform: Action=GetCallerIdentity&Version=2011-06-15
2016/10/31 18:01:05 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:06 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:06 [DEBUG] [aws-sdk-go] DEBUG: Response sts/GetCallerIdentity Details:
2016/10/31 18:01:06 [DEBUG] plugin: terraform: ---[ RESPONSE ]--------------------------------------
2016/10/31 18:01:06 [DEBUG] plugin: terraform: HTTP/1.1 200 OK
2016/10/31 18:01:06 [DEBUG] plugin: terraform: Connection: close
2016/10/31 18:01:06 [DEBUG] plugin: terraform: Content-Length: 406
2016/10/31 18:01:06 [DEBUG] plugin: terraform: Content-Type: text/xml
2016/10/31 18:01:06 [DEBUG] plugin: terraform: Date: Mon, 31 Oct 2016 09:01:05 GMT
2016/10/31 18:01:06 [DEBUG] plugin: terraform: X-Amzn-Requestid: 8f0a3081-9f48-11e6-a64d-b7a4dc898797
2016/10/31 18:01:06 [DEBUG] plugin: terraform: 
2016/10/31 18:01:06 [DEBUG] plugin: terraform: <GetCallerIdentityResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
2016/10/31 18:01:06 [DEBUG] plugin: terraform:   <GetCallerIdentityResult>
2016/10/31 18:01:06 [DEBUG] plugin: terraform:     <Arn>arn:aws:iam::481668315166:user/terraform</Arn>
2016/10/31 18:01:06 [DEBUG] plugin: terraform:     <UserId>AIDAI3G7OGSPZM4VRFGIM</UserId>
2016/10/31 18:01:06 [DEBUG] plugin: terraform:     <Account>481668315166</Account>
2016/10/31 18:01:06 [DEBUG] plugin: terraform:   </GetCallerIdentityResult>
2016/10/31 18:01:06 [DEBUG] plugin: terraform:   <ResponseMetadata>
2016/10/31 18:01:06 [DEBUG] plugin: terraform:     <RequestId>8f0a3081-9f48-11e6-a64d-b7a4dc898797</RequestId>
2016/10/31 18:01:06 [DEBUG] plugin: terraform:   </ResponseMetadata>
2016/10/31 18:01:06 [DEBUG] plugin: terraform: </GetCallerIdentityResponse>
2016/10/31 18:01:06 [DEBUG] plugin: terraform: 
2016/10/31 18:01:06 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:06 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:06 [DEBUG] Trying to get account ID via iam:GetUser
2016/10/31 18:01:06 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:06 [DEBUG] [aws-sdk-go] DEBUG: Request iam/GetUser Details:
2016/10/31 18:01:06 [DEBUG] plugin: terraform: ---[ REQUEST POST-SIGN ]-----------------------------
2016/10/31 18:01:06 [DEBUG] plugin: terraform: POST / HTTP/1.1
2016/10/31 18:01:06 [DEBUG] plugin: terraform: Host: iam.amazonaws.com
2016/10/31 18:01:06 [DEBUG] plugin: terraform: User-Agent: terraform/0.7.7 aws-sdk-go/1.4.14 (go1.7.1; darwin; amd64)
2016/10/31 18:01:06 [DEBUG] plugin: terraform: Content-Length: 33
2016/10/31 18:01:06 [DEBUG] plugin: terraform: Authorization: AWS4-HMAC-SHA256 Credential=AKIAIUVMU4M7NKKFBJBQ/20161031/us-east-1/iam/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date, Signature=7e7908210ca1beaaa2d1f097bb77d5e0324f2bed9cf52d74a1129b413d33355a
2016/10/31 18:01:06 [DEBUG] plugin: terraform: Content-Type: application/x-www-form-urlencoded; charset=utf-8
2016/10/31 18:01:06 [DEBUG] plugin: terraform: X-Amz-Date: 20161031T090106Z
2016/10/31 18:01:06 [DEBUG] plugin: terraform: Accept-Encoding: gzip
2016/10/31 18:01:06 [DEBUG] plugin: terraform: 
2016/10/31 18:01:06 [DEBUG] plugin: terraform: Action=GetUser&Version=2010-05-08
2016/10/31 18:01:06 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:08 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:08 [DEBUG] [aws-sdk-go] DEBUG: Response iam/GetUser Details:
2016/10/31 18:01:08 [DEBUG] plugin: terraform: ---[ RESPONSE ]--------------------------------------
2016/10/31 18:01:08 [DEBUG] plugin: terraform: HTTP/1.1 200 OK
2016/10/31 18:01:08 [DEBUG] plugin: terraform: Connection: close
2016/10/31 18:01:08 [DEBUG] plugin: terraform: Content-Length: 467
2016/10/31 18:01:08 [DEBUG] plugin: terraform: Content-Type: text/xml
2016/10/31 18:01:08 [DEBUG] plugin: terraform: Date: Mon, 31 Oct 2016 09:01:07 GMT
2016/10/31 18:01:08 [DEBUG] plugin: terraform: X-Amzn-Requestid: 8fcd3283-9f48-11e6-8cab-1bd11fd8e8dd
2016/10/31 18:01:08 [DEBUG] plugin: terraform: 
2016/10/31 18:01:08 [DEBUG] plugin: terraform: <GetUserResponse xmlns="https://iam.amazonaws.com/doc/2010-05-08/">
2016/10/31 18:01:08 [DEBUG] plugin: terraform:   <GetUserResult>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:     <User>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:       <Path>/</Path>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:       <UserName>terraform</UserName>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:       <Arn>arn:aws:iam::481668315166:user/terraform</Arn>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:       <UserId>AIDAI3G7OGSPZM4VRFGIM</UserId>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:       <CreateDate>2016-08-14T07:14:47Z</CreateDate>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:     </User>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:   </GetUserResult>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:   <ResponseMetadata>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:     <RequestId>8fcd3283-9f48-11e6-8cab-1bd11fd8e8dd</RequestId>
2016/10/31 18:01:08 [DEBUG] plugin: terraform:   </ResponseMetadata>
2016/10/31 18:01:08 [DEBUG] plugin: terraform: </GetUserResponse>
2016/10/31 18:01:08 [DEBUG] plugin: terraform: 
2016/10/31 18:01:08 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:08 [DEBUG] vertex provider.module.test_record.aws (close), got dep: module.test_record.provider.aws
2016/10/31 18:01:08 [DEBUG] vertex module.test_record.aws_route53_record.a_record, got dep: module.test_record.provider.aws
2016/10/31 18:01:08 [DEBUG] vertex module.test_record.aws_route53_record.cname_record (orphan), got dep: module.test_record.provider.aws
2016/10/31 18:01:08 [DEBUG] vertex root.module.test_record.aws_route53_record.a_record: walking
2016/10/31 18:01:08 [DEBUG] vertex root.module.test_record.aws_route53_record.a_record: evaluating
2016/10/31 18:01:08 [TRACE] [walkApply] Entering eval tree: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:08 [DEBUG] vertex root.module.test_record.aws_route53_record.cname_record (orphan): walking
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalInterpolate
2016/10/31 18:01:08 [DEBUG] vertex root.module.test_record.aws_route53_record.cname_record (orphan): evaluating
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalInstanceInfo
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalOpFilter
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalReadDiff
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalGetProvider
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalReadState
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalApply
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalCountFixZeroOneBoundary
2016/10/31 18:01:08 [DEBUG] vertex root.module.test_record.aws_route53_record.a_record: expanding/walking dynamic subgraph
2016/10/31 18:01:08 [DEBUG] apply: aws_route53_record.cname_record: executing Apply
2016/10/31 18:01:08 [DEBUG] vertex root.test_record.aws_route53_record.a_record: walking
2016/10/31 18:01:08 [DEBUG] vertex root.test_record.aws_route53_record.a_record: evaluating
2016/10/31 18:01:08 [TRACE] [walkApply] Entering eval tree: aws_route53_record.a_record
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalInstanceInfo
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalOpFilter
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalSequence
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalReadDiff
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalIf
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: terraform.EvalNoop
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalIf
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalInterpolate
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalGetProvider
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalReadState
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalValidateResource
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalDiff
2016/10/31 18:01:08 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:08 [DEBUG] [aws-sdk-go] DEBUG: Request route53/GetHostedZone Details:
2016/10/31 18:01:08 [DEBUG] plugin: terraform: ---[ REQUEST POST-SIGN ]-----------------------------
2016/10/31 18:01:08 [DEBUG] plugin: terraform: GET http://route53.amazonaws.com/2013-04-01/hostedzone/ZJWEBJ9GC1LO9 HTTP/1.1
2016/10/31 18:01:08 [DEBUG] plugin: terraform: Host: route53.amazonaws.com
2016/10/31 18:01:08 [DEBUG] plugin: terraform: User-Agent: terraform/0.7.7 aws-sdk-go/1.4.14 (go1.7.1; darwin; amd64)
2016/10/31 18:01:08 [DEBUG] plugin: terraform: Authorization: AWS4-HMAC-SHA256 Credential=AKIAIUVMU4M7NKKFBJBQ/20161031/us-east-1/route53/aws4_request, SignedHeaders=host;x-amz-date, Signature=718f06ef21f1538cabc531dcc1531f0143c4bdee11e72b129a9faa1c4da037c7
2016/10/31 18:01:08 [DEBUG] plugin: terraform: X-Amz-Date: 20161031T090108Z
2016/10/31 18:01:08 [DEBUG] plugin: terraform: Accept-Encoding: gzip
2016/10/31 18:01:08 [DEBUG] plugin: terraform: 
2016/10/31 18:01:08 [DEBUG] plugin: terraform: 
2016/10/31 18:01:08 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalReadDiff
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalCompareDiff
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalGetProvider
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalReadState
2016/10/31 18:01:08 [DEBUG] root.test_record: eval: *terraform.EvalApply
2016/10/31 18:01:08 [DEBUG] apply: aws_route53_record.a_record: executing Apply
2016/10/31 18:01:08 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:08 [DEBUG] [aws-sdk-go] DEBUG: Request route53/GetHostedZone Details:
2016/10/31 18:01:08 [DEBUG] plugin: terraform: ---[ REQUEST POST-SIGN ]-----------------------------
2016/10/31 18:01:08 [DEBUG] plugin: terraform: GET http://route53.amazonaws.com/2013-04-01/hostedzone/ZJWEBJ9GC1LO9 HTTP/1.1
2016/10/31 18:01:08 [DEBUG] plugin: terraform: Host: route53.amazonaws.com
2016/10/31 18:01:08 [DEBUG] plugin: terraform: User-Agent: terraform/0.7.7 aws-sdk-go/1.4.14 (go1.7.1; darwin; amd64)
2016/10/31 18:01:08 [DEBUG] plugin: terraform: Authorization: AWS4-HMAC-SHA256 Credential=AKIAIUVMU4M7NKKFBJBQ/20161031/us-east-1/route53/aws4_request, SignedHeaders=host;x-amz-date, Signature=718f06ef21f1538cabc531dcc1531f0143c4bdee11e72b129a9faa1c4da037c7
2016/10/31 18:01:08 [DEBUG] plugin: terraform: X-Amz-Date: 20161031T090108Z
2016/10/31 18:01:08 [DEBUG] plugin: terraform: Accept-Encoding: gzip
2016/10/31 18:01:08 [DEBUG] plugin: terraform: 
2016/10/31 18:01:08 [DEBUG] plugin: terraform: 
2016/10/31 18:01:08 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:09 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:09 [DEBUG] [aws-sdk-go] DEBUG: Response route53/GetHostedZone Details:
2016/10/31 18:01:09 [DEBUG] plugin: terraform: ---[ RESPONSE ]--------------------------------------
2016/10/31 18:01:09 [DEBUG] plugin: terraform: HTTP/1.1 200 OK
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Connection: close
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Content-Length: 728
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Content-Type: text/xml
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Date: Mon, 31 Oct 2016 09:01:08 GMT
2016/10/31 18:01:09 [DEBUG] plugin: terraform: X-Amzn-Requestid: 9084e9e2-9f48-11e6-a8da-a716d3c9a8e2
2016/10/31 18:01:09 [DEBUG] plugin: terraform: 
2016/10/31 18:01:09 [DEBUG] plugin: terraform: <?xml version="1.0"?>
2016/10/31 18:01:09 [DEBUG] plugin: terraform: <GetHostedZoneResponse xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><HostedZone><Id>/hostedzone/ZJWEBJ9GC1LO9</Id><Name>sl.internal.</Name><CallerReference>2016-09-07T11:13:51.721555234+09:00</CallerReference><Config><Comment>Managed by Terraform</Comment><PrivateZone>true</PrivateZone></Config><ResourceRecordSetCount>37</ResourceRecordSetCount></HostedZone><VPCs><VPC><VPCRegion>ap-northeast-1</VPCRegion><VPCId>vpc-1b28237e</VPCId></VPC><VPC><VPCRegion>ap-northeast-1</VPCRegion><VPCId>vpc-6a94a40f</VPCId></VPC><VPC><VPCRegion>ap-northeast-1</VPCRegion><VPCId>vpc-82d537e6</VPCId></VPC><VPC><VPCRegion>ap-northeast-1</VPCRegion><VPCId>vpc-9dd537f9</VPCId></VPC></VPCs></GetHostedZoneResponse>
2016/10/31 18:01:09 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:09 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:09 [DEBUG] Expanded record name: test_record.sl.internal
2016/10/31 18:01:09 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:09 [DEBUG] List resource records sets for zone: ZJWEBJ9GC1LO9, opts: {
2016/10/31 18:01:09 [DEBUG] plugin: terraform:   HostedZoneId: "ZJWEBJ9GC1LO9",
2016/10/31 18:01:09 [DEBUG] plugin: terraform:   StartRecordName: "test_record.sl.internal",
2016/10/31 18:01:09 [DEBUG] plugin: terraform:   StartRecordType: "CNAME"
2016/10/31 18:01:09 [DEBUG] plugin: terraform: }
2016/10/31 18:01:09 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:09 [DEBUG] [aws-sdk-go] DEBUG: Request route53/ListResourceRecordSets Details:
2016/10/31 18:01:09 [DEBUG] plugin: terraform: ---[ REQUEST POST-SIGN ]-----------------------------
2016/10/31 18:01:09 [DEBUG] plugin: terraform: GET http://route53.amazonaws.com/2013-04-01/hostedzone/ZJWEBJ9GC1LO9/rrset?name=test_record.sl.internal&type=CNAME HTTP/1.1
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Host: route53.amazonaws.com
2016/10/31 18:01:09 [DEBUG] plugin: terraform: User-Agent: terraform/0.7.7 aws-sdk-go/1.4.14 (go1.7.1; darwin; amd64)
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Authorization: AWS4-HMAC-SHA256 Credential=AKIAIUVMU4M7NKKFBJBQ/20161031/us-east-1/route53/aws4_request, SignedHeaders=host;x-amz-date, Signature=4e063de24b87038ff9520305bfad03be65c2ef36e8d2c295c347f211a5c83738
2016/10/31 18:01:09 [DEBUG] plugin: terraform: X-Amz-Date: 20161031T090109Z
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Accept-Encoding: gzip
2016/10/31 18:01:09 [DEBUG] plugin: terraform: 
2016/10/31 18:01:09 [DEBUG] plugin: terraform: 
2016/10/31 18:01:09 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:09 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:09 [DEBUG] [aws-sdk-go] DEBUG: Response route53/GetHostedZone Details:
2016/10/31 18:01:09 [DEBUG] plugin: terraform: ---[ RESPONSE ]--------------------------------------
2016/10/31 18:01:09 [DEBUG] plugin: terraform: HTTP/1.1 200 OK
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Connection: close
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Content-Length: 728
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Content-Type: text/xml
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Date: Mon, 31 Oct 2016 09:01:08 GMT
2016/10/31 18:01:09 [DEBUG] plugin: terraform: X-Amzn-Requestid: 90849b2a-9f48-11e6-8bca-8b85e7a18397
2016/10/31 18:01:09 [DEBUG] plugin: terraform: 
2016/10/31 18:01:09 [DEBUG] plugin: terraform: <?xml version="1.0"?>
2016/10/31 18:01:09 [DEBUG] plugin: terraform: <GetHostedZoneResponse xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><HostedZone><Id>/hostedzone/ZJWEBJ9GC1LO9</Id><Name>sl.internal.</Name><CallerReference>2016-09-07T11:13:51.721555234+09:00</CallerReference><Config><Comment>Managed by Terraform</Comment><PrivateZone>true</PrivateZone></Config><ResourceRecordSetCount>37</ResourceRecordSetCount></HostedZone><VPCs><VPC><VPCRegion>ap-northeast-1</VPCRegion><VPCId>vpc-1b28237e</VPCId></VPC><VPC><VPCRegion>ap-northeast-1</VPCRegion><VPCId>vpc-6a94a40f</VPCId></VPC><VPC><VPCRegion>ap-northeast-1</VPCRegion><VPCId>vpc-82d537e6</VPCId></VPC><VPC><VPCRegion>ap-northeast-1</VPCRegion><VPCId>vpc-9dd537f9</VPCId></VPC></VPCs></GetHostedZoneResponse>
2016/10/31 18:01:09 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:09 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:09 [DEBUG] Creating resource records for zone: ZJWEBJ9GC1LO9, name: test_record.sl.internal
2016/10/31 18:01:09 [DEBUG] plugin: terraform: 
2016/10/31 18:01:09 [DEBUG] plugin: terraform: {
2016/10/31 18:01:09 [DEBUG] plugin: terraform:   ChangeBatch: {
2016/10/31 18:01:09 [DEBUG] plugin: terraform:     Changes: [{
2016/10/31 18:01:09 [DEBUG] plugin: terraform:         Action: "UPSERT",
2016/10/31 18:01:09 [DEBUG] plugin: terraform:         ResourceRecordSet: {
2016/10/31 18:01:09 [DEBUG] plugin: terraform:           Name: "test_record.sl.internal",
2016/10/31 18:01:09 [DEBUG] plugin: terraform:           ResourceRecords: [{
2016/10/31 18:01:09 [DEBUG] plugin: terraform:               Value: "172.21.100.100"
2016/10/31 18:01:09 [DEBUG] plugin: terraform:             }],
2016/10/31 18:01:09 [DEBUG] plugin: terraform:           TTL: 300,
2016/10/31 18:01:09 [DEBUG] plugin: terraform:           Type: "A"
2016/10/31 18:01:09 [DEBUG] plugin: terraform:         }
2016/10/31 18:01:09 [DEBUG] plugin: terraform:       }],
2016/10/31 18:01:09 [DEBUG] plugin: terraform:     Comment: "Managed by Terraform"
2016/10/31 18:01:09 [DEBUG] plugin: terraform:   },
2016/10/31 18:01:09 [DEBUG] plugin: terraform:   HostedZoneId: "ZJWEBJ9GC1LO9"
2016/10/31 18:01:09 [DEBUG] plugin: terraform: }
2016/10/31 18:01:09 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:09 [DEBUG] Waiting for state to become: [accepted]
2016/10/31 18:01:09 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:09 [DEBUG] [aws-sdk-go] DEBUG: Request route53/ChangeResourceRecordSets Details:
2016/10/31 18:01:09 [DEBUG] plugin: terraform: ---[ REQUEST POST-SIGN ]-----------------------------
2016/10/31 18:01:09 [DEBUG] plugin: terraform: POST http://route53.amazonaws.com/2013-04-01/hostedzone/ZJWEBJ9GC1LO9/rrset/ HTTP/1.1
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Host: route53.amazonaws.com
2016/10/31 18:01:09 [DEBUG] plugin: terraform: User-Agent: terraform/0.7.7 aws-sdk-go/1.4.14 (go1.7.1; darwin; amd64)
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Content-Length: 446
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Authorization: AWS4-HMAC-SHA256 Credential=AKIAIUVMU4M7NKKFBJBQ/20161031/us-east-1/route53/aws4_request, SignedHeaders=content-length;host;x-amz-date, Signature=338d7a0bf069e5ec4c9885b36b3443776538a3f543a36231cda2d2a98815c1a2
2016/10/31 18:01:09 [DEBUG] plugin: terraform: X-Amz-Date: 20161031T090109Z
2016/10/31 18:01:09 [DEBUG] plugin: terraform: Accept-Encoding: gzip
2016/10/31 18:01:09 [DEBUG] plugin: terraform: 
2016/10/31 18:01:09 [DEBUG] plugin: terraform: <ChangeResourceRecordSetsRequest xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><ChangeBatch><Changes><Change><ResourceRecordSet><TTL>300</TTL><Type>A</Type><Name>test_record.sl.internal</Name><ResourceRecords><ResourceRecord><Value>172.21.100.100</Value></ResourceRecord></ResourceRecords></ResourceRecordSet><Action>UPSERT</Action></Change></Changes><Comment>Managed by Terraform</Comment></ChangeBatch></ChangeResourceRecordSetsRequest>
2016/10/31 18:01:09 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:10 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:10 [DEBUG] [aws-sdk-go] DEBUG: Response route53/ListResourceRecordSets Details:
2016/10/31 18:01:10 [DEBUG] plugin: terraform: ---[ RESPONSE ]--------------------------------------
2016/10/31 18:01:10 [DEBUG] plugin: terraform: HTTP/1.1 200 OK
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Connection: close
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Content-Length: 443
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Content-Type: text/xml
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Date: Mon, 31 Oct 2016 09:01:09 GMT
2016/10/31 18:01:10 [DEBUG] plugin: terraform: X-Amzn-Requestid: 9102a4af-9f48-11e6-b7ea-1b65fd468525
2016/10/31 18:01:10 [DEBUG] plugin: terraform: 
2016/10/31 18:01:10 [DEBUG] plugin: terraform: <?xml version="1.0"?>
2016/10/31 18:01:10 [DEBUG] plugin: terraform: <ListResourceRecordSetsResponse xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><ResourceRecordSets><ResourceRecordSet><Name>test_record.sl.internal.</Name><Type>CNAME</Type><TTL>300</TTL><ResourceRecords><ResourceRecord><Value>172.21.200.100</Value></ResourceRecord></ResourceRecords></ResourceRecordSet></ResourceRecordSets><IsTruncated>false</IsTruncated><MaxItems>100</MaxItems></ListResourceRecordSetsResponse>
2016/10/31 18:01:10 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:10 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:10 [DEBUG] Waiting for state to become: [accepted]
2016/10/31 18:01:10 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:10 [DEBUG] [aws-sdk-go] DEBUG: Request route53/ChangeResourceRecordSets Details:
2016/10/31 18:01:10 [DEBUG] plugin: terraform: ---[ REQUEST POST-SIGN ]-----------------------------
2016/10/31 18:01:10 [DEBUG] plugin: terraform: POST http://route53.amazonaws.com/2013-04-01/hostedzone/ZJWEBJ9GC1LO9/rrset/ HTTP/1.1
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Host: route53.amazonaws.com
2016/10/31 18:01:10 [DEBUG] plugin: terraform: User-Agent: terraform/0.7.7 aws-sdk-go/1.4.14 (go1.7.1; darwin; amd64)
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Content-Length: 451
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Authorization: AWS4-HMAC-SHA256 Credential=AKIAIUVMU4M7NKKFBJBQ/20161031/us-east-1/route53/aws4_request, SignedHeaders=content-length;host;x-amz-date, Signature=a46d0ea057d0376f161037a51c701511460cbf2f6fdc57f9e9c3ebdefa9b51e7
2016/10/31 18:01:10 [DEBUG] plugin: terraform: X-Amz-Date: 20161031T090110Z
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Accept-Encoding: gzip
2016/10/31 18:01:10 [DEBUG] plugin: terraform: 
2016/10/31 18:01:10 [DEBUG] plugin: terraform: <ChangeResourceRecordSetsRequest xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><ChangeBatch><Changes><Change><Action>DELETE</Action><ResourceRecordSet><Name>test_record.sl.internal.</Name><ResourceRecords><ResourceRecord><Value>172.21.200.100</Value></ResourceRecord></ResourceRecords><TTL>300</TTL><Type>CNAME</Type></ResourceRecordSet></Change></Changes><Comment>Deleted by Terraform</Comment></ChangeBatch></ChangeResourceRecordSetsRequest>
2016/10/31 18:01:10 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:10 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:10 [DEBUG] [aws-sdk-go] DEBUG: Response route53/ChangeResourceRecordSets Details:
2016/10/31 18:01:10 [DEBUG] plugin: terraform: ---[ RESPONSE ]--------------------------------------
2016/10/31 18:01:10 [DEBUG] plugin: terraform: HTTP/1.1 400 Bad Request
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Connection: close
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Content-Length: 423
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Content-Type: text/xml
2016/10/31 18:01:10 [DEBUG] plugin: terraform: Date: Mon, 31 Oct 2016 09:01:09 GMT
2016/10/31 18:01:10 [DEBUG] plugin: terraform: X-Amzn-Actiontrace: amzn1.tr.910c418b-9f48-11e6-b003-0a81ec150200.-1.7zlpwg.b6S4X5
2016/10/31 18:01:10 [DEBUG] plugin: terraform: X-Amzn-Requestid: 91047973-9f48-11e6-9bda-cf3f3e8e0891
2016/10/31 18:01:10 [DEBUG] plugin: terraform: 
2016/10/31 18:01:10 [DEBUG] plugin: terraform: <?xml version="1.0"?>
2016/10/31 18:01:10 [DEBUG] plugin: terraform: <ErrorResponse xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><Error><Type>Sender</Type><Code>InvalidChangeBatch</Code><Message>RRSet of type A with DNS name test_record.sl.internal. is not permitted because a conflicting RRSet of type  CNAME with the same DNS name already exists in zone sl.internal.</Message></Error><RequestId>91047973-9f48-11e6-9bda-cf3f3e8e0891</RequestId></ErrorResponse>
2016/10/31 18:01:10 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:10 [DEBUG] root.test_record: eval: *terraform.EvalWriteState
2016/10/31 18:01:10 [DEBUG] root.test_record: eval: *terraform.EvalApplyProvisioners
2016/10/31 18:01:10 [DEBUG] root.test_record: eval: *terraform.EvalIf
2016/10/31 18:01:10 [DEBUG] root.test_record: eval: *terraform.EvalWriteState
2016/10/31 18:01:10 [DEBUG] root.test_record: eval: *terraform.EvalWriteDiff
2016/10/31 18:01:10 [DEBUG] root.test_record: eval: *terraform.EvalApplyPost
2016/10/31 18:01:10 [ERROR] root.test_record: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* aws_route53_record.a_record: [ERR]: Error building changeset: InvalidChangeBatch: RRSet of type A with DNS name test_record.sl.internal. is not permitted because a conflicting RRSet of type  CNAME with the same DNS name already exists in zone sl.internal.
	status code: 400, request id: 91047973-9f48-11e6-9bda-cf3f3e8e0891
2016/10/31 18:01:10 [ERROR] root.test_record: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* aws_route53_record.a_record: [ERR]: Error building changeset: InvalidChangeBatch: RRSet of type A with DNS name test_record.sl.internal. is not permitted because a conflicting RRSet of type  CNAME with the same DNS name already exists in zone sl.internal.
	status code: 400, request id: 91047973-9f48-11e6-9bda-cf3f3e8e0891
2016/10/31 18:01:10 [ERROR] root.test_record: eval: *terraform.EvalOpFilter, err: 1 error(s) occurred:

* aws_route53_record.a_record: [ERR]: Error building changeset: InvalidChangeBatch: RRSet of type A with DNS name test_record.sl.internal. is not permitted because a conflicting RRSet of type  CNAME with the same DNS name already exists in zone sl.internal.
	status code: 400, request id: 91047973-9f48-11e6-9bda-cf3f3e8e0891
2016/10/31 18:01:10 [ERROR] root.test_record: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* aws_route53_record.a_record: [ERR]: Error building changeset: InvalidChangeBatch: RRSet of type A with DNS name test_record.sl.internal. is not permitted because a conflicting RRSet of type  CNAME with the same DNS name already exists in zone sl.internal.
	status code: 400, request id: 91047973-9f48-11e6-9bda-cf3f3e8e0891
2016/10/31 18:01:10 [DEBUG] vertex module.test_record.output.fqdn, got dep: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:10 [DEBUG] vertex root, waiting for: provider.module.test_record.aws (close)
2016/10/31 18:01:11 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:11 [DEBUG] [aws-sdk-go] DEBUG: Response route53/ChangeResourceRecordSets Details:
2016/10/31 18:01:11 [DEBUG] plugin: terraform: ---[ RESPONSE ]--------------------------------------
2016/10/31 18:01:11 [DEBUG] plugin: terraform: HTTP/1.1 200 OK
2016/10/31 18:01:11 [DEBUG] plugin: terraform: Connection: close
2016/10/31 18:01:11 [DEBUG] plugin: terraform: Content-Length: 315
2016/10/31 18:01:11 [DEBUG] plugin: terraform: Content-Type: text/xml
2016/10/31 18:01:11 [DEBUG] plugin: terraform: Date: Mon, 31 Oct 2016 09:01:10 GMT
2016/10/31 18:01:11 [DEBUG] plugin: terraform: X-Amzn-Requestid: 918e1c14-9f48-11e6-9bda-cf3f3e8e0891
2016/10/31 18:01:11 [DEBUG] plugin: terraform: 
2016/10/31 18:01:11 [DEBUG] plugin: terraform: <?xml version="1.0"?>
2016/10/31 18:01:11 [DEBUG] plugin: terraform: <ChangeResourceRecordSetsResponse xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><ChangeInfo><Id>/change/C1E505LKWZ04IM</Id><Status>PENDING</Status><SubmittedAt>2016-10-31T09:01:11.015Z</SubmittedAt><Comment>Deleted by Terraform</Comment></ChangeInfo></ChangeResourceRecordSetsResponse>
2016/10/31 18:01:11 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:11 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:11 [DEBUG] Waiting for state to become: [INSYNC]
2016/10/31 18:01:13 [DEBUG] vertex provider.module.test_record.aws (close), waiting for: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:15 [DEBUG] vertex root, waiting for: provider.module.test_record.aws (close)
2016/10/31 18:01:18 [DEBUG] vertex provider.module.test_record.aws (close), waiting for: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:20 [DEBUG] vertex root, waiting for: provider.module.test_record.aws (close)
2016/10/31 18:01:23 [DEBUG] vertex provider.module.test_record.aws (close), waiting for: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:25 [DEBUG] vertex root, waiting for: provider.module.test_record.aws (close)
2016/10/31 18:01:28 [DEBUG] vertex provider.module.test_record.aws (close), waiting for: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:30 [DEBUG] vertex root, waiting for: provider.module.test_record.aws (close)
2016/10/31 18:01:33 [DEBUG] vertex provider.module.test_record.aws (close), waiting for: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:35 [DEBUG] vertex root, waiting for: provider.module.test_record.aws (close)
2016/10/31 18:01:38 [DEBUG] vertex provider.module.test_record.aws (close), waiting for: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:40 [DEBUG] vertex root, waiting for: provider.module.test_record.aws (close)
2016/10/31 18:01:41 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:41 [DEBUG] [aws-sdk-go] DEBUG: Request route53/GetChange Details:
2016/10/31 18:01:41 [DEBUG] plugin: terraform: ---[ REQUEST POST-SIGN ]-----------------------------
2016/10/31 18:01:41 [DEBUG] plugin: terraform: GET http://route53.amazonaws.com/2013-04-01/change/C1E505LKWZ04IM HTTP/1.1
2016/10/31 18:01:41 [DEBUG] plugin: terraform: Host: route53.amazonaws.com
2016/10/31 18:01:41 [DEBUG] plugin: terraform: User-Agent: terraform/0.7.7 aws-sdk-go/1.4.14 (go1.7.1; darwin; amd64)
2016/10/31 18:01:41 [DEBUG] plugin: terraform: Authorization: AWS4-HMAC-SHA256 Credential=AKIAIUVMU4M7NKKFBJBQ/20161031/us-east-1/route53/aws4_request, SignedHeaders=host;x-amz-date, Signature=ac6ebf85b274f8d79cbc0a6e75991f29f82d42886de6641d7bcd170a1dfde015
2016/10/31 18:01:41 [DEBUG] plugin: terraform: X-Amz-Date: 20161031T090141Z
2016/10/31 18:01:41 [DEBUG] plugin: terraform: Accept-Encoding: gzip
2016/10/31 18:01:41 [DEBUG] plugin: terraform: 
2016/10/31 18:01:41 [DEBUG] plugin: terraform: 
2016/10/31 18:01:41 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:42 [DEBUG] plugin: terraform: aws-provider (internal) 2016/10/31 18:01:42 [DEBUG] [aws-sdk-go] DEBUG: Response route53/GetChange Details:
2016/10/31 18:01:42 [DEBUG] plugin: terraform: ---[ RESPONSE ]--------------------------------------
2016/10/31 18:01:42 [DEBUG] plugin: terraform: HTTP/1.1 200 OK
2016/10/31 18:01:42 [DEBUG] plugin: terraform: Connection: close
2016/10/31 18:01:42 [DEBUG] plugin: terraform: Content-Length: 284
2016/10/31 18:01:42 [DEBUG] plugin: terraform: Content-Type: text/xml
2016/10/31 18:01:42 [DEBUG] plugin: terraform: Date: Mon, 31 Oct 2016 09:01:41 GMT
2016/10/31 18:01:42 [DEBUG] plugin: terraform: X-Amzn-Requestid: a468ec0b-9f48-11e6-8d63-17e27ef28803
2016/10/31 18:01:42 [DEBUG] plugin: terraform: 
2016/10/31 18:01:42 [DEBUG] plugin: terraform: <?xml version="1.0"?>
2016/10/31 18:01:42 [DEBUG] plugin: terraform: <GetChangeResponse xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><ChangeInfo><Id>/change/C1E505LKWZ04IM</Id><Status>INSYNC</Status><SubmittedAt>2016-10-31T09:01:11.015Z</SubmittedAt><Comment>Deleted by Terraform</Comment></ChangeInfo></GetChangeResponse>
2016/10/31 18:01:42 [DEBUG] plugin: terraform: -----------------------------------------------------
2016/10/31 18:01:42 [DEBUG] root.test_record: eval: *terraform.EvalWriteState
2016/10/31 18:01:42 [DEBUG] root.test_record: eval: *terraform.EvalApplyPost
2016/10/31 18:01:42 [DEBUG] root.test_record: eval: *terraform.EvalUpdateStateHook
2016/10/31 18:01:42 [DEBUG] vertex provider.module.test_record.aws (close), got dep: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:42 [DEBUG] vertex provider.module.test_record.aws (close), got dep: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:42 [DEBUG] vertex root, got dep: provider.module.test_record.aws (close)
2016/10/31 18:01:42 [DEBUG] vertex root, got dep: module.test_record.aws_route53_record.cname_record (orphan)
2016/10/31 18:01:42 [DEBUG] vertex root, got dep: module.test_record.output.fqdn
2016/10/31 18:01:42 [DEBUG] vertex root, got dep: module.test_record.aws_route53_record.a_record
2016/10/31 18:01:42 [DEBUG] vertex root, got dep: module.test_record.plan-destroy
2016/10/31 18:01:42 [DEBUG] plugin: waiting for all plugin processes to complete...
2016/10/31 18:01:42 [DEBUG] plugin: /usr/local/bin/terraform: plugin process exited
```

--- a/builtin/providers/aws/resource_aws_route53_record.go
+++ b/builtin/providers/aws/resource_aws_route53_record.go
@@ -49,7 +49,6 @@ func resourceAwsRoute53Record() *schema.Resource {
 			"type": &schema.Schema{
 				Type:         schema.TypeString,
 				Required:     true,
-				ForceNew:     true,
 				ValidateFunc: validateRoute53RecordType,
 			},
 
@@ -81,7 +80,6 @@ func resourceAwsRoute53Record() *schema.Resource {
 			"set_identifier": &schema.Schema{
 				Type:     schema.TypeString,
 				Optional: true,
-				ForceNew: true,
 			},
 
 			"alias": &schema.Schema{
@@ -223,13 +221,124 @@ func resourceAwsRoute53RecordUpdate(d *schema.ResourceData, meta interface{}) er
 	// Route 53 supports CREATE, DELETE, and UPSERT actions. We use UPSERT, and
 	// AWS dynamically determines if a record should be created or updated.
 	// Amazon Route 53 can update an existing resource record set only when all
-	// of the following values match: Name, Type
-	// (and SetIdentifier, which we don't use yet).
-	// See http://docs.aws.amazon.com/Route53/latest/APIReference/API_ChangeResourceRecordSets_Requests.html#change-rrsets-request-action
-	//
-	// Because we use UPSERT, for resouce update here we simply fall through to
-	// our resource create function.
-	return resourceAwsRoute53RecordCreate(d, meta)
+	// of the following values match: Name, Type and SetIdentifier
+	// See http://docs.aws.amazon.com/Route53/latest/APIReference/API_ChangeResourceRecordSets.html
+
+	if !d.HasChange("type") && !d.HasChange("set_identifier") {
+		// If neither type nor set_identifier changed we use UPSERT,
+		// for resouce update here we simply fall through to
+		// our resource create function.
+		return resourceAwsRoute53RecordCreate(d, meta)
+	}
+
+	// Otherwise we delete the existing record and create a new record within
+	// a transactional change
+	conn := meta.(*AWSClient).r53conn
+	zone := cleanZoneID(d.Get("zone_id").(string))
+
+	var err error
+	zoneRecord, err := conn.GetHostedZone(&route53.GetHostedZoneInput{Id: aws.String(zone)})
+	if err != nil {
+		return err
+	}
+	if zoneRecord.HostedZone == nil {
+		return fmt.Errorf("[WARN] No Route53 Zone found for id (%s)", zone)
+	}
+
+	// Build the to be deleted record
+	en := expandRecordName(d.Get("name").(string), *zoneRecord.HostedZone.Name)
+	typeo, _ := d.GetChange("type")
+
+	oldRec := &route53.ResourceRecordSet{
+		Name: aws.String(en),
+		Type: aws.String(typeo.(string)),
+	}
+
+	if v, _ := d.GetChange("ttl"); v.(int) != 0 {
+		oldRec.TTL = aws.Int64(int64(v.(int)))
+	}
+
+	// Resource records
+	if v, _ := d.GetChange("records"); v != nil {
+		recs := v.(*schema.Set).List()
+		if len(recs) > 0 {
+			oldRec.ResourceRecords = expandResourceRecords(recs, typeo.(string))
+		}
+	}
+
+	// Alias record
+	if v, _ := d.GetChange("alias"); v != nil {
+		aliases := v.(*schema.Set).List()
+		if len(aliases) == 1 {
+			alias := aliases[0].(map[string]interface{})
+			oldRec.AliasTarget = &route53.AliasTarget{
+				DNSName:              aws.String(alias["name"].(string)),
+				EvaluateTargetHealth: aws.Bool(alias["evaluate_target_health"].(bool)),
+				HostedZoneId:         aws.String(alias["zone_id"].(string)),
+			}
+		}
+	}
+
+	if v, _ := d.GetChange("set_identifier"); v.(string) != "" {
+		oldRec.SetIdentifier = aws.String(v.(string))
+	}
+
+	// Build the to be created record
+	rec, err := resourceAwsRoute53RecordBuildSet(d, *zoneRecord.HostedZone.Name)
+	if err != nil {
+		return err
+	}
+
+	// Delete the old and create the new records in a single batch. We abuse
+	// StateChangeConf for this to retry for us since Route53 sometimes returns
+	// errors about another operation happening at the same time.
+	changeBatch := &route53.ChangeBatch{
+		Comment: aws.String("Managed by Terraform"),
+		Changes: []*route53.Change{
+			&route53.Change{
+				Action:            aws.String("DELETE"),
+				ResourceRecordSet: oldRec,
+			},
+			&route53.Change{
+				Action:            aws.String("CREATE"),
+				ResourceRecordSet: rec,
+			},
+		},
+	}
+
+	req := &route53.ChangeResourceRecordSetsInput{
+		HostedZoneId: aws.String(cleanZoneID(*zoneRecord.HostedZone.Id)),
+		ChangeBatch:  changeBatch,
+	}
+
+	log.Printf("[DEBUG] Updating resource records for zone: %s, name: %s\n\n%s",
+		zone, *rec.Name, req)
+
+	respRaw, err := changeRoute53RecordSet(conn, req)
+	if err != nil {
+		return errwrap.Wrapf("[ERR]: Error building changeset: {{err}}", err)
+	}
+
+	changeInfo := respRaw.(*route53.ChangeResourceRecordSetsOutput).ChangeInfo
+
+	// Generate an ID
+	vars := []string{
+		zone,
+		strings.ToLower(d.Get("name").(string)),
+		d.Get("type").(string),
+	}
+	if v, ok := d.GetOk("set_identifier"); ok {
+		vars = append(vars, v.(string))
+	}
+
+	d.SetId(strings.Join(vars, "_"))
+
+	err = waitForRoute53RecordSetToSync(conn, cleanChangeID(*changeInfo.Id))
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsRoute53RecordRead(d, meta)
 }
 
 func resourceAwsRoute53RecordCreate(d *schema.ResourceData, meta interface{}) error {

--- a/builtin/providers/aws/resource_aws_route53_record_test.go
+++ b/builtin/providers/aws/resource_aws_route53_record_test.go
@@ -339,6 +339,56 @@ func TestAccAWSRoute53Record_TypeChange(t *testing.T) {
 	})
 }
 
+func TestAccAWSRoute53Record_SetIdentiferChange(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_route53_record.basic_to_weighted",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckRoute53RecordDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccRoute53RecordSetIdentifierChangePre,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRoute53RecordExists("aws_route53_record.basic_to_weighted"),
+				),
+			},
+
+			// Cause a change, which will trigger a refresh
+			resource.TestStep{
+				Config: testAccRoute53RecordSetIdentifierChangePost,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRoute53RecordExists("aws_route53_record.basic_to_weighted"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSRoute53Record_AliasChange(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_route53_record.elb_alias_change",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckRoute53RecordDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccRoute53RecordAliasChangePre,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRoute53RecordExists("aws_route53_record.elb_alias_change"),
+				),
+			},
+
+			// Cause a change, which will trigger a refresh
+			resource.TestStep{
+				Config: testAccRoute53RecordAliasChangePost,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRoute53RecordExists("aws_route53_record.elb_alias_change"),
+				),
+			},
+		},
+	})
+}
+
 func TestAccAWSRoute53Record_empty(t *testing.T) {
 	resource.Test(t, resource.TestCase{
 		PreCheck:      func() { testAccPreCheck(t) },
@@ -1019,6 +1069,82 @@ resource "aws_route53_record" "sample" {
   type = "A"
   ttl = "30"
   records = ["127.0.0.1", "8.8.8.8"]
+}
+`
+
+const testAccRoute53RecordSetIdentifierChangePre = `
+resource "aws_route53_zone" "main" {
+	name = "notexample.com"
+}
+
+resource "aws_route53_record" "basic_to_weighted" {
+  zone_id = "${aws_route53_zone.main.zone_id}"
+  name = "sample"
+  type = "A"
+  ttl = "30"
+  records = ["127.0.0.1", "8.8.8.8"]
+}
+`
+
+const testAccRoute53RecordSetIdentifierChangePost = `
+resource "aws_route53_zone" "main" {
+	name = "notexample.com"
+}
+
+resource "aws_route53_record" "basic_to_weighted" {
+  zone_id = "${aws_route53_zone.main.zone_id}"
+  name = "sample"
+  type = "A"
+  ttl = "30"
+  records = ["127.0.0.1", "8.8.8.8"]
+  set_identifier = "cluster-a"
+  weighted_routing_policy {
+    weight = 100
+  }
+}
+`
+
+const testAccRoute53RecordAliasChangePre = `
+resource "aws_route53_zone" "main" {
+	name = "notexample.com"
+}
+
+resource "aws_elb" "alias_change" {
+  name = "foobar-tf-elb-alias-change"
+  availability_zones = ["us-west-2a"]
+
+  listener {
+    instance_port = 80
+    instance_protocol = "http"
+    lb_port = 80
+    lb_protocol = "http"
+  }
+}
+
+resource "aws_route53_record" "elb_alias_change" {
+  zone_id = "${aws_route53_zone.main.zone_id}"
+  name = "alias-change"
+  type = "A"
+
+  alias {
+    zone_id = "${aws_elb.alias_change.zone_id}"
+    name = "${aws_elb.alias_change.dns_name}"
+    evaluate_target_health = true
+  }
+}
+`
+
+const testAccRoute53RecordAliasChangePost = `
+resource "aws_route53_zone" "main" {
+	name = "notexample.com"
+}
+
+resource "aws_route53_record" "elb_alias_change" {
+  zone_id = "${aws_route53_zone.main.zone_id}"
+  name = "alias-change"
+  type = "CNAME"
+  ttl = "30"
+  records = ["www.terraform.io"]
 }
 `
 

